lib/spack/spack/cmd/graph.py

```diff
@@ -41,36 +48,46 @@ def setup_parser(subparser):
     )
 
     subparser.add_argument(
-        "-i",
-        "--installed",
-        action="store_true",
-        help="graph installed specs, or specs in the active env (implies --dot)",
+        "-i", "--installed", action="store_true", help="graph specs from the DB"
     )
 
     arguments.add_common_arguments(subparser, ["deptype", "specs"])
 
 
 def graph(parser, args):
-    if args.installed and args.specs:
-        tty.die("cannot specify specs with --installed")
+    env = ev.active_environment()
+    if args.installed and env:
+        tty.die("cannot use --installed with an active environment")
+
+    if args.installed and args.ascii:
+        tty.die("cannot use --installed with --ascii")
+
+    if env and args.ascii:
+        tty.die("cannot use --ascii with an active environment")
```

Comment:
Why?

Reply:
Not sure, previously `--installed` implied `--dot` I didn't touch that but can have a look if this is a real constraint.

Reply:
Limitation removed